clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1064,6 +1058,17 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
     }
   }
 
+  if (getLangOpts().SYCLIsDevice) {
```

Comment:
Is this right here?  If we have the attribute, I would think we would want to apply this to the function, right?  There is no reason to limit this to just SYCL as it is an FPGA attribute.

Reply:
>>There is no reason to limit this to just SYCL as it is an FPGA attribute.

Agreed. I have removed the check.